frontend/src/providers/ContextProvider.tsx

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Meta Platforms, Inc. and affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable react-refresh/only-export-components */
+
+import { createContext, ReactNode, useContext, useState } from "react";
+
+interface AppContextType {
+  error: any;
+  setError: (e: any) => void;
+  zoomHandlers: any;
+  setZoomHandlers: (e: any) => void;
+}
+
+const AppContext = createContext<AppContextType | undefined>(undefined);
+
+export const ContextProvider = ({ children }: { children: ReactNode }) => {
```

Comment:
using a ContextProvider is fine, but if you're interested we've been using zustand to share state between components. It's been much easier to work with: https://github.com/pmndrs/zustand.